ofrak_core/test_ofrak/components/test_gzip_component.py

```diff
@@ -1,87 +1,103 @@
         assert patched_decompressed_data == self.EXPECTED_REPACKED_DATA
 
 
-class TestGzipUnpackWithTrailingBytes(UnpackModifyPackPattern):
-    EXPECTED_TAG = GzipData
-    INITIAL_DATA = b"Hello World"
-    EXPECTED_DATA = INITIAL_DATA  # Change expected when modifier is created
-    INNER_FILENAME = "hello.bin"
-    GZIP_FILENAME = "hello.bin.gz"
-
-    async def create_root_resource(self, ofrak_context: OFRAKContext) -> Resource:
-        with tempfile.TemporaryDirectory() as d:
-            file_path = os.path.join(d, self.INNER_FILENAME)
-            with open(file_path, "wb") as f:
-                f.write(self.INITIAL_DATA)
-
-            gzip_path = os.path.join(d, self.GZIP_FILENAME)
-            gzip_command = ["pigz", file_path]
-            subprocess.run(gzip_command, check=True, capture_output=True)
-
-            # Add trailing bytes
-            with open(gzip_path, "ab") as a:
-                a.write(b"\xDE\xAD\xBE\xEF")
-                a.close()
-            return await ofrak_context.create_root_resource_from_file(gzip_path)
-
-    async def unpack(self, root_resource: Resource) -> None:
-        await root_resource.unpack_recursively()
-
-    async def modify(self, root_resource: Resource) -> None:
-        pass
-
-    async def repack(self, root_resource: Resource) -> None:
-        pass
-
-    async def verify(self, root_resource: Resource) -> None:
-        gzip_data = await root_resource.get_data()
-        with tempfile.TemporaryDirectory() as d:
-            gzip_path = os.path.join(d, self.GZIP_FILENAME)
-            with open(gzip_path, "wb") as f:
-                f.write(gzip_data)
-
-            gunzip_command = ["pigz", "-d", "-c", gzip_path]
-            try:
-                result = subprocess.run(gunzip_command, check=True, capture_output=True)
-                data = result.stdout
-            except subprocess.CalledProcessError as e:
-                if e.returncode == 2 or e.returncode == -2:
-                    data = e.stdout
-                else:
-                    raise
-
-            assert data == self.EXPECTED_DATA
+class TestGzipUnpackModifyPack(GzipUnpackModifyPackPattern):
+    def write_gzip(self, gzip_path: Path):
+        with gzip.GzipFile(gzip_path, mode="w") as gzip_file:
+            gzip_file.write(self.INITIAL_DATA)
+
+
+class TestGzipWithMultipleMembersUnpackModifyPack(GzipUnpackModifyPackPattern):
+    def write_gzip(self, gzip_path: Path):
+        with gzip.GzipFile(gzip_path, mode="w") as gzip_file:
+            middle = len(self.INITIAL_DATA) // 2
+            gzip_file.write(self.INITIAL_DATA[:middle])
+            gzip_file.write(self.INITIAL_DATA[middle:])
+
+
+class TestGzipWithTrailingBytesUnpackModifyPack(GzipUnpackModifyPackPattern):
+    def write_gzip(self, gzip_path: Path):
+        with gzip.GzipFile(gzip_path, mode="w") as gzip_file:
+            gzip_file.write(self.INITIAL_DATA)
```

Comment:
Could you just concatenate DEADBEEF on this line?

Reply:
no, that would add it to the compressed data section instead of writing extra trailing bytes.